### PR TITLE
feat: enable Prometheus endpoint for external provisioner

### LINKED ISSUE #12128: Display build status via Prometheus
A customer reported they were unable to get build status (e.g. "failed," "succeeded") via Promethus metrics and had to script with our audit API to get this detail.

Use case: "how often are developer builds failing"

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/12128

This PR enables Prometheus metrics endpoint for a standalone provisionerd instance.

_This could be a missight or bug as we advertise in golden files that Prometheus is already supported by provisioner: [link](https://github.com/coder/coder/blob/392fecee87ae1cde210ed38a31ab8c62ed325a77/helm/provisioner/tests/testdata/default_values.golden#L108)._

### CODE REVIEW

File: enterprise/cli/provisionerdaemons.go [hunk 1]
```diff
@@ -291,6 +315,13 @@ func (r *RootCmd) provisionerDaemonStart() *clibase.Cmd {
 			Value:       clibase.StringArrayOf(&logFilter),
 			Default:     "",
 		},
+		{
+			Flag:        "prometheus-address",
+			Env:         "CODER_PROMETHEUS_ADDRESS",
+			Description: "The bind address to serve prometheus metrics.",
+			Value:       clibase.StringOf(&prometheusAddress),
+			Default:     "127.0.0.1:2112",
+		},
```
Comment (johnstcn) [inline enterprise/cli/provisionerdaemons.go:324]: non-blocking: I'm in two minds as to whether it should be `CODER_PROVISIONERD_PROMETHEUS_ADDRESS` for consistency with other provisionerd env vars, or just `CODER_PROMETHEUS_ADDRESS` for consistency with coderd.

**Review by johnstcn — APPROVED** [review]

File: enterprise/cli/provisionerdaemons.go (hunk 1 above)
Comment (mtojek) [reply enterprise/cli/provisionerdaemons.go:324]: I had this concern too, but then noticed that we already "support" `CODER_PROMETHEUS_ADDRESS` everywhere, so extra +1 point for backward compatibility even though it didn't work before?

**Comment by ericpaulsen** [comment]
closes #10601

File: enterprise/cli/provisionerdaemons.go (hunk 1 above)
Comment (johnstcn) [reply enterprise/cli/provisionerdaemons.go:324]: I'm sold. :shipit:

**Comment by mtojek** [comment]
Technically we focus on provisioner here, but we can focus on workspace proxy in the follow-up 👍

**Pull request merged**